service/internal/proctelemetry/process_telemetry.go

```diff
@@ -31,100 +48,224 @@ type processMetrics struct {
 
 	// mu protects everything bellow.
 	mu         sync.Mutex
 	lastMsRead time.Time
 	ms         *runtime.MemStats
 }
 
-// RegisterProcessMetrics creates a new set of processMetrics (mem, cpu) that can be used to measure
-// basic information about this process.
-func RegisterProcessMetrics(registry *metric.Registry, ballastSizeBytes uint64) error {
+func NewProcessMetrics(logger *zap.Logger, registry *featuregate.Registry, meter otelmetric.MeterProvider, ballastSizeBytes uint64) *processMetrics {
 	pm := &processMetrics{
 		startTimeUnixNano: time.Now().UnixNano(),
 		ballastSizeBytes:  ballastSizeBytes,
 		ms:                &runtime.MemStats{},
+		useOtelForMetrics: registry.IsEnabled(obsreportconfig.UseOtelForInternalMetricsfeatureGateID),
+		logger:            logger,
 	}
+
+	if pm.useOtelForMetrics {
+		pm.meter = meter.Meter(scopeName)
+	}
+
+	return pm
+}
+
+// RegisterProcessMetrics creates a new set of processMetrics (mem, cpu) that can be used to measure
+// basic information about this process.
+func (pm *processMetrics) RegisterProcessMetrics(ctx context.Context, ocRegistry *ocmetric.Registry) error {
 	var err error
 	pm.proc, err = process.NewProcess(int32(os.Getpid()))
 	if err != nil {
 		return err
 	}
 
+	if pm.useOtelForMetrics {
+		err = pm.createOtelMetrics()
+		if err != nil {
+			return err
+		}
+
+		err = pm.recordWithOtel(ctx)
+		if err != nil {
+			return err
+		}
+	} else { // record with OC
+		err = pm.createOCMetrics(ocRegistry)
+		if err != nil {
+			return err
+		}
+
+		err = pm.recordWithOC()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (pm *processMetrics) createOtelMetrics() error {
+	var errors, err error
+
+	pm.otelProcessUptime, err = pm.meter.AsyncFloat64().Counter(
+		"process_uptime",
+		instrument.WithDescription("Uptime of the process"),
+		instrument.WithUnit(unit.Milliseconds))
+	errors = multierr.Append(errors, err)
+
+	pm.otelAllocMem, err = pm.meter.AsyncInt64().Counter(
+		"process_runtime_heap_alloc_bytes",
+		instrument.WithDescription("Bytes of allocated heap objects (see 'go doc runtime.MemStats.HeapAlloc')"),
+		instrument.WithUnit(unit.Bytes))
+	errors = multierr.Append(errors, err)
+
+	pm.otelTotalAllocMem, err = pm.meter.AsyncInt64().Counter(
+		"process_runtime_total_alloc_bytes",
+		instrument.WithDescription("Cumulative bytes allocated for heap objects (see 'go doc runtime.MemStats.TotalAlloc')"),
+		instrument.WithUnit(unit.Bytes))
+	errors = multierr.Append(errors, err)
+
+	pm.otelSysMem, err = pm.meter.AsyncInt64().Counter(
+		"process_runtime_total_sys_memory_bytes",
+		instrument.WithDescription("Total bytes of memory obtained from the OS (see 'go doc runtime.MemStats.Sys')"),
+		instrument.WithUnit(unit.Bytes))
+	errors = multierr.Append(errors, err)
+
+	pm.otelCpuSeconds, err = pm.meter.AsyncFloat64().Counter(
+		"process_cpu_seconds",
+		instrument.WithDescription("Total CPU user and system time in seconds"),
+		instrument.WithUnit(unit.Milliseconds))
```

Comment:
`instrument.WithUnit()` takes in a [unit.Unit](https://pkg.go.dev/go.opentelemetry.io/otel/metric@v0.34.0/unit#Unit) but the only provided unit for time is Millisecond.  The name of this metric is `process_cpu_seconds` so I'm wondering if I should rename this or if the metric name needs to be consistent with the OC metric name?

Reply:
I think for just leave out the unit, and add a comment about it. It is better to not have an unit than having the wrong one.

Reply:
Actually, since Otel follows UCUM, [seconds should be represented as](https://unitsofmeasure.org/ucum#section-Base-Units): `s`. I think you should be able to do a `unit.Unit("s")`. 

We should ask otel-go to expand these as well.

Reply:
Thanks for the suggestion! I am now using `unit.Unit("s")` Do you know how I can confirm that this worked? The output from prometheus doesn't seem to include the unit

e.g.
```
# HELP otelcol_process_uptime_total Uptime of the process
# TYPE otelcol_process_uptime_total counter
otelcol_process_uptime_total 10.295116
```

Reply:
Oh, you're right. The otel-go sdk doesn't export units yet, but I think that is cool for now.